lang_qc/models/qc_flow_status.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2022 Genome Research Ltd.
+#
+# Author: Marina Gourtovaia <mg8@sanger.ac.uk>
+#
+# This file is part of npg_langqc.
+#
+# npg_langqc is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum, unique
+from typing import List
+
+from pydantic import BaseModel, Field
+
+
+class QcFlowStatus(BaseModel):
+    """
+    A representation of a single QC flow status.
+
+    The QC flow status migh encapsulate mulriple QC states. It might
+    correspond to an unknown (not yet assigned) QC state.
+    """
+
+    label: str = Field(title="A human readable label of the status")
+    param: str = Field(
+        title="A query parameter that should be used retrieve "
+        + "wells at this stage of the flow"
+    )
+
+
+@unique
+class QcFlowStatusEnum(str, Enum):
+    """
+    An enumeration of known QC flow states. The order of the statuses is
+    consistent with the temporal flow of the manual QC process.
+    """
+
+    INBOX = "inbox"
```

Comment:
Are we to expect any substates here? e.g. QC complete, failed because of bad metric. Or will I be able to see both state and justification on the per-well QC outcome? Or can I forget about groupings entirely and leave it up to the API to give me correctly grouped data?

Reply:
These statuses correspond to stages of the QC process. Some of them will have wells in multiple QC states. The QC states will be displayed next to individual wells. The frontend should rely on the API to return correct lists of states.